specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
+                    length and no more than 8 octets in length.
+    Definition:     A Signed Integer stores an integer (meaning that it
+                    can be written without a fractional component) which
+                    may be negative, positive, or zero. Because EBML
+                    limits Signed Integers to 8 octets in length a
+                    Signed Element may store a number from
+                    −9,223,372,036,854,775,808 to
+                    +9,223,372,036,854,775,807.
+
+Element Data Type:   Unsigned Integer
+
+    Endianness:     Big-endian
+    Length:         An Unsigned Integer Element MUST at least 1 octet in
+                    length and no more than 8 octets in length.
+    Definition:     An Unsigned Integer stores an integer (meaning that
+                    it can be written without a fractional component)
+                    which may be positive or zero. Because EBML limits
+                    Unsigned Integers to 8 octets in length an unsigned
+                    Element may store a number from 0 to
+                    18,446,744,073,709,551,615.
+
+Element Data Type:   Float
+
+    Endianness:     Big-endian
+    Length:         A Float Element MUST either 4 octets (32 bit) or 8
+                    octets (64 bit) in length.
+    Definition:     A Float Elements stores a floating-point number as
+                    defined in IEEE 754.
+
+Element Data Type:   String
+
+    Endianness:     None
+    Length:         A String Element may use any length (included zero)
+                    up to the maximum Element Data Size value permitted.
+    Definition:     A String Element may either be empty (zero-length)
+                    or contain Printable ASCII characters in the range
+                    of 0x20 to 0x7E. Octets with all bits set to zero
```

Comment:
Note: some controversy about why zero-padding is allowed, but all prior docs seem consistent about allowing it.